pkg/serializer/serializer.go

```diff
@@ -329,6 +330,15 @@ func (s *Serializer) SendIterableSeries(series marshaler.IterableMarshaler) erro
 	return s.Forwarder.SubmitSeries(seriesPayloads, extraHeaders)
 }
 
+// IsIterableSeriesSupported returns whether `SendIterableSeries` is supported.
+// Should be removed when `serializePayloadJSON` (useV1API && !s.enableJSONStream) will be removed
+func (s *Serializer) IsIterableSeriesSupported() bool {
+	useV1API := !config.Datadog.GetBool("use_v2_api.series")
```

Comment:
I'm not a huge fan of using negated logic like this. I'd rather have something like:

```golang
useV2API := config.Datadog.GetBool("use_v2_api.series")

return useV2API || s.enableJSONStream
```